subsys/bluetooth/controller/ll_sw/ull_adv.c

```diff
@@ -1569,6 +1573,13 @@ static int init_reset(void)
 		lll_adv_data_init(&ll_adv[handle].lll.scan_rsp);
 	}
 
+	/* Make sure that set #0 is initialized with empty legacy PDUs. This is
+	 * especially important if legacy HCI interface is used for advertising
+	 * because it allows to enable advertising without any configuration,
+	 * thus we need to have PDUs already initialized.
+	 */
+	init_legacy_pdu(&ll_adv[0]);
+
```

Comment:
```suggestion
	
```

Instead do the common initialization of bit fields and len here: https://github.com/zephyrproject-rtos/zephyr/blob/master/subsys/bluetooth/controller/ll_sw/nordic/lll/lll_adv.c#L150
And, set the `type` after the call to `lll_adv_data_init` here: https://github.com/zephyrproject-rtos/zephyr/blob/master/subsys/bluetooth/controller/ll_sw/ull_adv.c#L1568

Reply:
@cvinayak This would do a lot of redundant initialization. `lll_adv_data_init` is also used for aux and sync which initialize PDU from scratch anyway. Also we don't really need to initialize anything other that set #0, because those sets will be initialized on create. The only reason we need to initialize set #0 is because legacy advertising can be started by legacy HCI command without any prior configuration.

So I'd keep this initialization separately. I can just rework it a bit into smth like `lll_adv_pdu_init` and call it for `&ll_adv[0].lll.adv_data` and `&ll_adv[0].lll.scan_rsp`. Alternatively, initialization can be postponed until we switch to legacy HCI commands when 1st legacy command is used.